src/submission/repositories/xpub-submission-root.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import * as Knex from 'knex';
+import { InfraLogger as logger } from '../../logger';
+import { SubmissionId } from '../types';
+import { SubmissionRepository, SubmissionDTO } from './types';
+
+type entryMeta = {
+    articleType: string;
+    title: string;
+};
+// this is the xpub shema type
+type DatabaseEntry = {
+    id: SubmissionId;
+    updated: Date;
+    created_by: string;
+    status: string;
+    meta: entryMeta;
+};
+
+export default class XpubSubmissionRootRepository implements SubmissionRepository {
+    private readonly TABLE_NAME = 'manuscript';
+
+    public constructor(private readonly knex: Knex<{}, unknown[]>) {}
+
+    close(): void {
+        logger.log(`Closing KnexSubmissionRepository.`);
+        this.knex.destroy();
+    }
+
+    public async findAll(): Promise<SubmissionDTO[]> {
+        const result = await this.knex
+            .withSchema('public')
+            .select<DatabaseEntry[]>('id', 'updated', 'created_by', 'status', 'meta')
+            .from(this.TABLE_NAME);
+
+        return result.map(this.entryToDTO);
+    }
+
+    public async findById(id: SubmissionId): Promise<SubmissionDTO> {
+        const rows = await this.knex
+            .withSchema('public')
+            .select<DatabaseEntry[]>('id', 'updated', 'created_by', 'status', 'meta')
+            .from(this.TABLE_NAME)
+            .where({ id });
+
+        return this.entryToDTO(rows[0]);
```

Comment:
`return this.entryToDTO(rows[0]);`
Will this line throw an exception when entryToDTO tries to destructure if the select returns an empty array?

Reply:
``` 
it('No results - DOES THIS BREAK???', async (): Promise<void> => {
            mockKnex.where = jest.fn().mockReturnValue([]);
            const repo = new XpubSubmissionRootRepository((mockKnex as unknown) as Knex);
            const result = await repo.findById(entryId);
            expect((result as SubmissionDTO).id).toEqual(databaseEntries[0].id);
});
```

Adding this test does break its during the destructure